File: python/cell.py
class Cell:
    row = 0
    column = 0
    north = None
    south = None
    east = None
    west = None

    def __init__(self, row, column):
        self.row = row
        self.column = column
        self.links = {}

    def link(self, cell, bidi=True):
        self.links[cell] = True
        if bidi:
            cell.link(self, False)
        return self

    def unlink(self, cell, bidi=True):
        self.links[cell] = False
        if bidi:
            cell.unlink(self, False)
        return self

    def links(self):
        return self.links.keys()

    def linked(self, cell):
        return self.links.get(cell, False)

    def neighbors(self):
        li = []
        if self.north:
            li.append(self.north)
        if self.south:
            li.append(self.south)
        if self.east:
            li.append(self.east)
        if self.west:
            li.append(self.west)
        return li

    

File: go/maze/cell.go
package maze

type Cell struct {
	row    int
	column int
	north  *Cell
	south  *Cell
	east   *Cell
	west   *Cell
	links  map[*Cell]bool
}

func newCell(row int, col int) *Cell {
	cell := Cell{row: row, column: col}
	cell.links = make(map[*Cell]bool)
	return &cell
}

func (cell *Cell) Link(target *Cell, bidi bool) {
	cell.links[target] = true

	if bidi {
		target.Link(cell, false)
	}
}

func (cell *Cell) Unlink(target *Cell, bidi bool) {
	delete(cell.links, target)

	if bidi {
		target.Unlink(cell, false)
	}
}

func (cell *Cell) linked(target *Cell) bool {
	_, ok := cell.links[target]

	if !ok {
		return false
	}

	return true
}

func (cell *Cell) neighbors() []*Cell {
	var neighbors []*Cell

	if cell.north != nil {
		neighbors = append(neighbors, cell.north)
	}

	if cell.south != nil {
		neighbors = append(neighbors, cell.south)
	}

	if cell.east != nil {
		neighbors = append(neighbors, cell.east)
	}

	if cell.west != nil {
		neighbors = append(neighbors, cell.west)
	}

	return neighbors
}
